pkg/repository/v1/sqlcv1/payload-store.sql

```diff
@@ -0,0 +1,153 @@
+-- name: ReadPayload :one
+SELECT *
+FROM v1_payload
+WHERE
+    tenant_id = @tenantId::UUID
+    AND type = @type::v1_payload_type
+    AND id = @id::BIGINT
+    AND inserted_at = @insertedAt::TIMESTAMPTZ
+;
+
+-- name: ReadPayloads :many
+WITH inputs AS (
+    SELECT
+        UNNEST(@ids::BIGINT[]) AS id,
+        UNNEST(@insertedAts::TIMESTAMPTZ[]) AS inserted_at,
+        UNNEST(@tenantIds::UUID[]) AS tenant_id,
+        UNNEST(CAST(@types::TEXT[] AS v1_payload_type[])) AS type
+)
+
+SELECT *
+FROM v1_payload
+WHERE (tenant_id, id, inserted_at, type) IN (
+        SELECT tenant_id, id, inserted_at, type
+        FROM inputs
+    )
+;
+
+-- name: WritePayloads :exec
+WITH inputs AS (
+    SELECT
+        UNNEST(@ids::BIGINT[]) AS id,
+        UNNEST(@insertedAts::TIMESTAMPTZ[]) AS inserted_at,
+        UNNEST(CAST(@types::TEXT[] AS v1_payload_type[])) AS type,
+        UNNEST(CAST(@locations::TEXT[] AS v1_payload_location[])) AS location,
+        UNNEST(@externalLocationKeys::TEXT[]) AS external_location_key,
+        UNNEST(@inlineContents::JSONB[]) AS inline_content,
+        UNNEST(@tenantIds::UUID[]) AS tenant_id
+)
+INSERT INTO v1_payload (
+    tenant_id,
+    id,
+    inserted_at,
+    type,
+    location,
+    external_location_key,
+    inline_content
+)
+
+SELECT
+    i.tenant_id,
+    i.id,
+    i.inserted_at,
+    i.type,
+    i.location,
+    CASE WHEN i.external_location_key = '' OR i.location = 'EXTERNAL' THEN NULL ELSE i.external_location_key END,
```

Comment:
any thoughts on this? I kinda prefer it to needing the overwrite, since the key will never be empty